### ISSUE TITLE
Promise.all chaining is broken

### ISSUE THREAD

**Issue by SaschaNaz** [Microsoft/TypeScript#8007]
**TypeScript Version:**  1.8.29.0 (default for Visual Studio 2015 Update 2)

**Code**

```ts
// worked on TS 1.8.4.0
let promises: Promise<number>[];
let promiseall: Promise<number[]> = Promise.all(promises)
```

**Expected behavior:** No compilation error. 

**Actual behavior:** 

```
Type 'Promise<Promise<number>[]>' is not assignable to type 'Promise<number[]>'.
  Type 'Promise<number>[]' is not assignable to type 'number[]'.
    Type 'Promise<number>' is not assignable to type 'number'.
```

**Comment by SaschaNaz** [comment]
Duplicate of #7566 :/
